autogluon/task/tabular_prediction/predictor.py

```diff
@@ -688,6 +688,75 @@ def load_data_internal(self, dataset='train', return_X=True, return_y=True):
         y = load_y() if return_y else None
         return X, y
 
+    def reduce_memory_size(self, remove_data=True, remove_fit_stack=True, requires_save=True, reduce_children=False):
```

Comment:
should `reduce_memory_size()` have another arg like:
`only_keep_topmodel`

that if True calls `delete_models()` on all the models unused by best_model

Reply:
I think we should keep reduce_memory_size functionality to be unrelated to inference. Removing models impacts inference time, whereas reduce_memory_size is only related to auxiliary information. I am planning to have a more comprehensive minimize function in a later PR that will do what you are suggesting.